expr/parse.go

```diff
@@ -115,12 +115,27 @@ type MetricRequest struct {
 	Until  int32
 }
 
+// ParseContext includes contextual information that affects how the current
+// expression is parsed.
+// This should only be set by internal calls.
+type ParseContext struct {
```

Comment:
if internal only, we should not export this type

Reply:
This is used as a parameter to the exported `Parse()` function,which is why I decided to export this type. The comment `// This should only be set by internal calls` was copied from the previous description for the `piped` parameter of the `Parse()` function. 

It is odd to have parameters that should only be set for internal calls though. I think a better way to write this is to have an exported `Parse(expr string)` function which calls an unexported `parse(expr string, pc parseContext)` function. WDYT? Only problem I see is that if there are Metrictank consumers that are using the current `Parse()` function and for some reason need to set the `piped` variable.

Reply:
makes sense. Your alternative suggestion is reasonable but not clearly better. We can just keep what you have.